src/main/java/com/solace/samples/TopicPublisher.java

```diff
@@ -46,59 +45,63 @@
  */
 public class TopicPublisher {
 
-    private static final Logger LOG = LogManager.getLogger(TopicPublisher.class.getName());
+    final String TOPIC_NAME = "T/GettingStarted/pubsub";
 
-    // connectionfactory.solaceConnectionLookup in file "jndi.properties"
     final String SOLACE_CONNECTION_LOOKUP = "solaceConnectionLookup";
-    // topic.topicLookup in file "jndi.properties"
-    final String TOPIC_LOOKUP = "topicLookup";
-
-    // session parameters
-    final int ACK_MODE = Session.AUTO_ACKNOWLEDGE;
-    final boolean IS_TRANSACTED = false;
-
-    private void run() {
-        try {
-            // pick up properties from the "jndi.properties" file
-            Context initialContext = new InitialContext();
-            TopicConnectionFactory factory = (TopicConnectionFactory) initialContext.lookup(SOLACE_CONNECTION_LOOKUP);
```

Comment:
Lets not do a jndi lookup in this example... Just do the JNDI lookup in the QueueConsumerJNDI.java and QueueProducerJNDI.java... Instead use: https://github.com/apache/qpid-jms/blob/master/qpid-jms-client/src/main/java/org/apache/qpid/jms/JmsConnectionFactory.java